staging/src/k8s.io/apimachinery/pkg/util/clock/clock.go

```diff
@@ -66,7 +66,12 @@ func NewFakeClock(t time.Time) *testclocks.FakeClock {
 // IntervalClock implements Clock, but each invocation of Now steps
 // the clock forward the specified duration.
 //
-// Deprecated: Use k8s.io/utils/clock/testing.IntervalClock instead.
+// WARNING: most of the Clock methods just `panic`;
+// only PassiveClock is honestly implemented.
+// The alternative, SimpleIntervalClock, has only the
+// PassiveClock methods.
+//
+// Deprecated: Use k8s.io/utils/clock/testing.SimpleIntervalClock instead.
```

Comment:
While most methods of `IntervalClock` just `panic`, I was wondering if we should recommend `SimpleIntervalClock` before actually deprecating `IntervalClock` in `k8s.io/utils` because we want `k8s.io/utils/clock` to be a pure replacement for now. The end goal is the same - to use `SimpleIntervalClock`, but do you think we should first deprecate `IntervalClock` and then make this change?

Reply:
I think both "deprecate" and "recommend" are just forms of pressure that we can exert on maintainers, and I see no reason to not do them both now.

There are two IntervalClocks, one in k/apimachinery and one in k/utils; both should be deprecated with a recommendation to use SimpleIntervalClock instead.